### ISSUE TITLE
iaPreciseTouchDescriptor.bin is no longer available in the new firmware

### ISSUE THREAD

**Issue by cydia2020** [ipts-linux-org/ipts-linux-new#4]
the binary iaPreciseTouchDescriptor.bin which is necessary is no longer available in the new Surface Book firmware. cannot find it anymore, and if i grab one from other machine, it simply wont work

**Comment by ardacoskunses** [comment]
Hello,

I do not have a SurfaceBook to try this but here is a suggestion:
Did you try getting all other firmwares from other machine and load all of them?

Descriptor.bin: A HID descriptor file provided by vendor. The HID Driver appends the panel’s HID descriptor to information.
KernelSKL.bin: Touch Vendor provided OpenCL kernel includes touch algorithms.
SFTConfig.bin: Touch Vendor provided configuration binary.
and the one you already got:
iaPreciseTouchDescriptor.bin: A HID descriptor file. A bare minimum HID descriptor descripting the device. The HID Driver appends the panel’s HID descriptor to information.

As long as HW configuration does not change this should work.
Firmwares must be from same SurfaceBook, SP4 version would not work on SurfaceBook

**Comment by cydia2020** [comment]
No, I tried that from another old version SB device, but that ain't working

**Comment by ardacoskunses** [comment]
Could you share your kernel log?

**Comment by pabloinigoblasco** [comment]
I also have this problem, testing SFP4:

xxx@xxx:/lib/firmware/intel/ipts$ ls -l
total 7828
-rwxr-xr-x 1 root root    2647 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0076.bin
-rwxr-xr-x 1 root root    2647 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0078.bin
-rwxr-xr-x 1 root root    2741 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0079.bin
-rwxr-xr-x 1 root root    2008 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0101.bin
-rwxr-xr-x 1 root root    2008 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0102.bin
-rwxr-xr-x 1 root root    2647 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0103.bin
-rwxr-xr-x 1 root root    2031 Jul  1 22:04 SurfaceTouchServicingDescriptorMSHW0137.bin
-rwxr-xr-x 1 root root 1133592 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0079.bin
-rwxr-xr-x 1 root root     516 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0079.bin.sig
-rwxr-xr-x 1 root root 1299992 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0101.bin
-rwxr-xr-x 1 root root     516 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0101.bin.sig
-rwxr-xr-x 1 root root 1299992 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0102.bin
-rwxr-xr-x 1 root root     516 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0102.bin.sig
-rwxr-xr-x 1 root root 1299820 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0137.bin
-rwxr-xr-x 1 root root     516 Jul  1 22:04 SurfaceTouchServicingKernelMSHW0137.bin.sig
-rwxr-xr-x 1 root root  922012 Jul  1 22:04 SurfaceTouchServicingKernelSKLMSHW0076.bin
-rwxr-xr-x 1 root root  927796 Jul  1 22:04 SurfaceTouchServicingKernelSKLMSHW0078.bin
-rwxr-xr-x 1 root root  927796 Jul  1 22:04 SurfaceTouchServicingKernelSKLMSHW0103.bin
-rwxr-xr-x 1 root root   15466 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0076.bin
-rwxr-xr-x 1 root root   15478 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0078.bin
-rwxr-xr-x 1 root root   11166 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0079.bin
-rwxr-xr-x 1 root root   11662 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0101.bin
-rwxr-xr-x 1 root root   11662 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0102.bin
-rwxr-xr-x 1 root root   15478 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0103.bin
-rwxr-xr-x 1 root root   11662 Jul  1 22:04 SurfaceTouchServicingSFTConfigMSHW0137.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0076.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0078.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0079.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0101.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0102.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0103.bin
-rwxr-xr-x 1 root root     256 Jul  1 22:04 SurfaceTouchServicingTouchBlobMSHW0137.bin
**lrwxrwxrwx 1 root root      53 Jul  1 22:22 integ_descriptor.bin -> /lib/firmware/intel/ipts/iaPreciseTouchDescriptor.bin [BROKEN LINK]** 
lrwxrwxrwx 1 root root      67 Jul  1 22:04 integ_sft_cfg_skl.bin -> /lib/firmware/intel/ipts/SurfaceTouchServicingSFTConfigMSHW0078.bin
-rw-r--r-- 1 root root     708 Jul  1 22:05 ipts_fw_config.bin
lrwxrwxrwx 1 root root      68 Jul  1 22:04 vendor_descriptor.bin -> /lib/firmware/intel/ipts/SurfaceTouchServicingDescriptorMSHW0078.bin
lrwxrwxrwx 1 root root      67 Jul  1 22:04 vendor_kernel_skl.bin -> /lib/firmware/intel/ipts/SurfaceTouchServicingKernelSKLMSHW0078.bin

**Comment by pabloinigoblasco** [comment]
$ sudo dmesg | grep ipts

> [    2.594555] >> let init ipts
> [    7.140632] IPTS ipts_mei_cl_init() is called
> [    7.146586] ipts mei::3e8d0870-271a-4208-8eb5-9acb9402ae04:0F: Direct firmware load for intel/ipts/intel_desc.bin failed with error -2
> [    7.146590] ipts mei::3e8d0870-271a-4208-8eb5-9acb9402ae04:0F: can't add hid device: -5
> [    7.146596] ipts mei::3e8d0870-271a-4208-8eb5-9acb9402ae04:0F: error in handling resp msg
> [   16.654385] Modules linked in: rfcomm fuse bnep btusb btrtl btbcm btintel bluetooth sd_mod sg intel_ipts i2c_designware_platform 
> 

....

**Comment by ardacoskunses** [comment]
Is it possible if you try the file from older versions?
If it is not working with old file, share the logs please.
Otherwise logs only indicate the missing file, which is expected.

**Comment by CaiMiao** [comment]
https://www.reddit.com/r/SurfaceLinux/comments/55bfq2/touch_firmware_files_for_sp4_touch_screen/d89bvc2/

Here are old files of Pro4 shared on reddit, it may work for Book, need someone test it.
